pf/tfbridge/provider_read.go

```diff
@@ -141,22 +141,17 @@ func (p *provider) readResource(
 		return plugin.ReadResult{}, nil
 	}
 
-	// TF interpretes a null new state as an indication that the resource does not
-	// exist in the cloud provider.
-	newStateNull, err := resp.NewState.IsNull()
```

Comment:
Previously IsNull check was done on the DynamicValue domain, and after the change it's done on the tftypes.Value domain. There's a very slim chance the issue was there, namely that there were values that return IsNull() = false on DynamicValue domain that then translate to Null values on tftypes.Value domain.